### PR TITLE
Fix fetchAnnotations for sandboxed Tasks

### PR DESCRIPTION
**Pull request by vlam21**
MTurk HIT IDs are assigned to Tasks when submitting them to MTurk, so sandboxed Tasks (not sent to MTurk) have NULL mturk_hit. This CL includes a missing check for this in fetchAnnotations so that we don't do a lookup to MTurk for Task status using a NULL mturk_hit.

### CODE REVIEW

**Comment by echen** [comment]
Do tasks sent to the MTurk Sandbox not get HIT IDs? (I thought they did.) Or do you mean another kind of sandbox?

File: lib/daemons/human_eval_thrift_daemon.rb [hunk 1]
```diff
@@ -61,21 +61,29 @@ def fetchAnnotations(fetch_annotation_params)
...
       raise HumanEvalException.new("No Task exists with the given Task ID: #{task_id}") if task.nil?
 
-      assignment = MTurkUtils.fetch_assignment_for_task(task)
-      if assignment.nil?
-        task_result.status = TaskStatus::PENDING
+      # Check if the Task has no registered MTurk HIT ID. It doesn't make sense to do an MTurk
+      # lookup for a Task that doesn't have one, since it's the key used to do the lookup.
+      # A Task doesn't have an MTurk HIT ID if it was created through the Thrift API and was added
+      # to a sandboxed evaluation.
+      if task.mturk_hit == nil
```
Comment (echen) [inline lib/daemons/human_eval_thrift_daemon.rb]: minor: `task.mturk_hit.nil?`

**Comment by vlam21** [comment]
I realized I'm unclear on what a sandboxed/non-prod evaluation actually means; this whole time I thought it was an abstraction introduced by Clockwork Raven. Do tasks belonging to sandboxed evaluations get sent to MTurk? But just don't get shown to MTurkers?

**Comment by echen** [comment]
No, it's not an abstraction introduced by CR. Amazon provides both "Production" and "Sandbox" functionalities. The Sandbox is essentially a development environment that provides the same functionally as Production, but doesn't charge for anything. https://requestersandbox.mturk.com/developer/sandbox

Aside: I'm not actually sure we want to keep the Sandbox functionality in CR. It's a relatively small amount of code, and the main benefit is in case you want to run tasks that will be judged internally (and so you don't want to pay Amazon anything), but it seems kind of buggy. However, being able to do internal judgments for free is useful, and building this functionality into CR would add just that much more code to be maintained (though it would certainly be nice, and I don't think it would be that much work).

**Comment by echen** [comment]
To more clearly answer your question: yes, tasks get sent to MTurk. Regular MTurk workers can, in theory, see tasks in the Sandbox as well (though they'll almost certainly never check it, since they won't get paid for doing any work). If you form a custom qualification (like the Trusted workers we have) in Production, though, I don't think you can directly transfer it to the Sandbox -- you need to recreate the qual in Sandbox itself.

**Pull request merged**